src/Common/BSONCXXHelper.h

```diff
@@ -23,13 +27,43 @@ using bsoncxx::builder::basic::document;
 using bsoncxx::builder::basic::kvp;
 using bsoncxx::builder::basic::make_document;
 
-static bsoncxx::types::bson_value::value fieldAsBSONValue(const Field & field, const DataTypePtr & type)
+static bsoncxx::types::bson_value::value tryOid(const std::string & str)
+{
+    try
+    {
+        return bsoncxx::oid(str);
+    } catch (bsoncxx::v_noabi::exception & e)
+    {
+        if (e.code() == bsoncxx::v_noabi::error_code::k_invalid_oid)
+            return bsoncxx::types::b_string{str};
+        throw;
+    }
+}
+
+static bsoncxx::types::bson_value::value fieldAsBSONValue(const Field & field, const DataTypePtr & type, const bool & shouldTryOid = false)
```

Comment:
"shouldTryOid" sounds too generic

Reply:
Also, there is no need to pass a bool by const reference, you can pass it by value.